server/src/openfga-yaml-schema.ts

```diff
@@ -1,4 +1,348 @@
+      const userTypes = type.metadata.relations[tuple.relation].directly_related_user_types;
+
+      // Found no direct relationship
+      if (!userTypes?.filter(t => t.type === user && t.relation === relation && !t.wildcard).length) {
+        validateTuple.errors.push(userNotTypeRestriction(`${user}#${relation}`, tuple, instancePath));
+        return false;
+      }
+
+      // Found no matching condition
+      if (tuple.condition && conditions) {
+        if (!userTypes?.filter(t => t.type === user && t.relation === relation && !t.wildcard && t.condition === tuple.condition?.name).length) {
+          validateTuple.errors.push(conditionNotTypeRestriction(user, tuple, instancePath));
+          return false;
+        }
+
+        // Check parameters for matching condition
+        return validateConditionParams(tuple, tuple.condition.name, conditions, instancePath);
+      }
+    }
+  } else {
+    const user =  tuple.user.split(":")[0];
+    const object = tuple.object.split(":")[0];
+    const type = typeDefs.filter(t => t.type === object)[0];
+
+    // Get all direct relationships
+    if (type.metadata?.relations && type.metadata?.relations[tuple.relation]) {
+      const userTypes = type.metadata.relations[tuple.relation].directly_related_user_types;
+
+      // Found no direct relationship
+      if (!userTypes?.filter(t => t.type === user && !t.relation && !t.wildcard).length) {
+        validateTuple.errors.push(userNotTypeRestriction(user, tuple, instancePath));
+        return false;
+      }
+
+      // Found no matching condition
+      if (tuple.condition && conditions) {
+        if (!userTypes?.filter(t => t.type === user && !t.relation && !t.wildcard && t.condition === tuple.condition?.name).length) {
+          validateTuple.errors.push(conditionNotTypeRestriction(user, tuple, instancePath));
+          return false;
+        }
+
+        // Check parameters for matching condition
+        return validateConditionParams(tuple, tuple.condition.name, conditions, instancePath);
+      }
+    }
+  }
+  // Encountered no errors in validation
+  return true;
+}
+
+function validateConditionExists(tuple: Tuple, conditions: { [key: string]: Condition; } | undefined, instancePath: string): boolean {
+  if (!tuple.condition) {
+    return true;
+  }
+
+  validateTuple.errors = validateTuple.errors || [];
+
+  // Condition on tuple not found
+  if (!conditions || !conditions[tuple.condition.name]) {
+    validateTuple.errors.push(conditionDoesntExist(tuple, instancePath));
+    return false;
+  }
+  return true;
+}
+
+function validateConditionParams(
+  tuple: Tuple,
+  condition: string,
+  conditions: { [key: string]: Condition; },
+  instancePath: string): boolean {
+
+  validateTuple.errors = validateTuple.errors || [];
+
+  if (tuple.condition && tuple.condition.context && conditions[condition].parameters) {
+    for (const p of Object.keys(tuple.condition.context)) {
```

Comment:
```suggestion
    for (const param of Object.keys(tuple.condition.context)) {
```

Please don't use single letter variables 🙏🏽